go/ai/generate.go

```diff
@@ -86,8 +86,170 @@ func LookupModel(provider, name string) *Model {
 	return (*Model)(core.LookupActionFor[*GenerateRequest, *GenerateResponse, *GenerateResponseChunk](atype.Model, provider, name))
 }
 
-// Generate applies the [Model] to some input, handling tool requests.
-func (m *Model) Generate(ctx context.Context, req *GenerateRequest, cb ModelStreamingCallback) (*GenerateResponse, error) {
+type GenerateParams struct {
+	Request      *GenerateRequest
+	Stream       ModelStreamingCallback
+	History      []*Message
+	SystemPrompt *Message
+}
+
+// GenerateParamsBuilder adds.
```

Comment:
Finish doc.

Also, we would typically call this type `GenerateOption`.
Standard pattern: https://pkg.go.dev/google.golang.org/api/option#ClientOption

Reply:
Done.